runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/ProcessBundleDescriptors.java

```diff
@@ -210,13 +215,18 @@ private static String uniquifyId(String idBase, Predicate<String> idUsed) {
 
   private static Coder<WindowedValue<?>> instantiateWireCoder(
       RemoteGrpcPort port, Map<String, RunnerApi.Coder> components) throws IOException {
-    MessageWithComponents byteArrayCoder =
-        LengthPrefixUnknownCoders.forCoder(
-            port.getCoderId(), Components.newBuilder().putAllCoders(components).build(), true);
+//    MessageWithComponents byteArrayCoder =
+//        LengthPrefixUnknownCoders.forCoder(
+//            port.getCoderId(), Components.newBuilder().putAllCoders(components).build(), true);
+//    Coder<?> javaCoder =
+//        CoderTranslation.fromProto(
+//            byteArrayCoder.getCoder(),
+//            RehydratedComponents.forComponents(byteArrayCoder.getComponents()));
     Coder<?> javaCoder =
         CoderTranslation.fromProto(
-            byteArrayCoder.getCoder(),
-            RehydratedComponents.forComponents(byteArrayCoder.getComponents()));
+            components.get(port.getCoderId()),
+            RehydratedComponents.forComponents(
+                Components.newBuilder().putAllCoders(components).build()));
```

Comment:
Are we throwing away changes to the components here or just RehydratedComponents only used for reading?

Reply:
Ah, yes, this one is tricky. The problem is that the code that I commented out is deserialising stuff that we get over the gRCP bridge (from the harness) using byte array coders (we're using `KvCoder<LenghtPrefixCoder<ByteArrayCoder>, LenghtPrefixCoder<ByteArrayCoder>>` instead of `KvCoder<LenghtPrefixCoder<StringUtf8Coder>, LenghtPrefixCoder<StringUtf8Coder>>` which Flink will then try to use to send data across the network after the pruning functions. The proper solution is to use byte arrays everywhere on the Flink Runner side.

Reply:
Adding a todo. The commit for this has `(HACK)` in the title... 😉